aws/sdk-codegen/src/main/kotlin/software/amazon/smithy/rustsdk/AwsCustomizableOperationDecorator.kt

```diff
@@ -0,0 +1,101 @@
+/*
+ * Copyright Amazon.com, Inc. or its affiliates. All Rights Reserved.
+ * SPDX-License-Identifier: Apache-2.0
+ */
+
+package software.amazon.smithy.rustsdk
+
+import software.amazon.smithy.rust.codegen.client.smithy.generators.client.CustomizableOperationCustomization
+import software.amazon.smithy.rust.codegen.client.smithy.generators.client.CustomizableOperationSection
+import software.amazon.smithy.rust.codegen.core.rustlang.CargoDependency
+import software.amazon.smithy.rust.codegen.core.rustlang.Writable
+import software.amazon.smithy.rust.codegen.core.rustlang.rust
+import software.amazon.smithy.rust.codegen.core.rustlang.rustTemplate
+import software.amazon.smithy.rust.codegen.core.rustlang.writable
+import software.amazon.smithy.rust.codegen.core.smithy.RuntimeConfig
+import software.amazon.smithy.rust.codegen.core.smithy.RuntimeType
+
+class CustomizableOperationTestHelpers(runtimeConfig: RuntimeConfig) :
+    CustomizableOperationCustomization() {
+    private val codegenScope = arrayOf(
+        *RuntimeType.preludeScope,
+        "AwsUserAgent" to AwsRuntimeType.awsHttp(runtimeConfig)
+            .resolve("user_agent::AwsUserAgent"),
+        "BeforeTransmit" to RuntimeType.smithyRuntimeApi(runtimeConfig)
+            .resolve("client::interceptors::context::phase::BeforeTransmit"),
+        "ConfigBag" to RuntimeType.smithyRuntimeApi(runtimeConfig).resolve("config_bag::ConfigBag"),
+        "ConfigBagAccessors" to RuntimeType.smithyRuntimeApi(runtimeConfig).resolve("client::orchestrator::ConfigBagAccessors"),
+        "InterceptorContext" to RuntimeType.smithyRuntimeApi(runtimeConfig)
+            .resolve("client::interceptors::InterceptorContext"),
+        "RequestTime" to RuntimeType.smithyRuntimeApi(runtimeConfig).resolve("client::orchestrator::RequestTime"),
+        "SharedInterceptor" to RuntimeType.smithyRuntimeApi(runtimeConfig)
+            .resolve("client::interceptors::SharedInterceptor"),
+        "TestParamsSetterInterceptor" to CargoDependency.smithyRuntime(runtimeConfig).withFeature("test-util")
+            .toType().resolve("client::test_util::interceptor::TestParamsSetterInterceptor"),
+    )
+
+    override fun section(section: CustomizableOperationSection): Writable =
+        writable {
+            if (section is CustomizableOperationSection.CustomizableOperationImpl) {
+                if (section.operationShape == null) {
```

Comment:
Let's add a `TODO(enableNewSmithyRuntime)` comment here to delete this branch when removing middleware.

Reply:
Added in 3074cdb.